### PR TITLE
Migrate addon links to TS

### PR DESCRIPTION
**Pull request by sairus2k**
Issue: #5030

## What I did

Migrate @storybook/addon-links to TypeScript

### CODE REVIEW

**Comment by kroeder** [comment]
Hey, thanks a lot for starting another TS migration! Do you need any help? 🙂

**Review by ndelangen — APPROVED** [review]

**Comment by ndelangen** [comment]
Seems there are some issues here:
```
ERR! => Failed to build the preview
ERR! /tmp/storybook/examples/angular-cli/src/stories/addon-links.stories.ts
ERR! ERROR in /tmp/storybook/examples/angular-cli/src/stories/addon-links.stories.ts(9,14):
ERR! TS2554: Expected 2 arguments, but got 1.
ERR! /tmp/storybook/examples/angular-cli/src/stories/index.ts
```

File: addons/links/src/preview.ts [hunk 1]
```diff
@@ -14,23 +21,24 @@ const generateUrl = id => {
...
-export const hrefTo = (kind, name) =>
+export const hrefTo = (kind: string, name: string): Promise<string> =>
   new Promise(resolve => {
     resolve(generateUrl(toId(kind, name)));
   });
 
-const linksListener = e => {
-  const { sbKind: kind, sbStory: story } = e.target.dataset;
+const linksListener = (e: SyntheticEvent<HTMLLinkElement>) => {
+  const { sbKind: kind, sbStory: story } = e.currentTarget.dataset;
```
Comment (Hypnosphi) [inline addons/links/src/preview.ts]: `currentTarget` is always `document` which makes it throw on each click, see e.g.

https://next--storybooks-html.netlify.com/?path=/story/welcome--welcome

File: addons/links/src/react/components/link.tsx [hunk 2]
```diff
@@ -9,48 +8,58 @@ import { navigate, hrefTo } from '../../preview';
...
-      href: '/',
-    };
+interface State {
+  href: string;
+}
 
-    this.handleClick = this.handleClick.bind(this);
-  }
+export default class LinkTo extends PureComponent<Props, State> {
+  defaultProps: Props = {
```
Comment (Hypnosphi) [inline addons/links/src/react/components/link.tsx]: forgot `static`?

**Pull request merged**